s3/replication/common/src/s3replicationcommon/s3_get_object_tagging.py

```diff
@@ -0,0 +1,148 @@
+#
+# Copyright (c) 2021 Seagate Technology LLC and/or its Affiliates
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+import aiohttp
+import re
+import sys
+import urllib
+from defusedxml.ElementTree import fromstring
+from s3replicationcommon.aws_v4_signer import AWSV4Signer
+from s3replicationcommon.log import fmt_reqid_log
+from s3replicationcommon.s3_common import S3RequestState
+from s3replicationcommon.timer import Timer
+
+
+class S3AsyncGetObjectTagging:
+    def __init__(self, session, request_id,
+                 bucket_name, object_name):
+        """Initialise."""
+        self._session = session
+        # Request id for better logging.
+        self._request_id = request_id
+        self._logger = session.logger
+
+        self._bucket_name = bucket_name
+        self._object_name = object_name
+
+        self._remote_down = False
+        self._http_status = None
+
+        self._timer = Timer()
+        self._state = S3RequestState.INITIALISED
+
+    def get_state(self):
+        """Returns current request state."""
+        return self._state
+
+    def get_execution_time(self):
+        """Return total time for GET operation."""
+        return self._timer.elapsed_time_ms()
+
+    def get_tags_count(self):
+        """Returns tags count."""
+        return len(self._response_tags_dict)
+
+    def get_tags_dict(self):
+        """Returns tags dictionary."""
+        return self._response_tags_dict
+
+    def get_tags_value(self, key):
+        """Returns the value for the given key."""
+        return self._response_tags_dict[key]
```

Comment:
If the key does not exist, what will happen?

Reply:
code updated.
done.